tornado/test/httpclient_test.py

```diff
@@ -138,6 +138,16 @@ def get(self):
             self.set_header(k, v)
 
 
+class InvalidGzipHandler(RequestHandler):
+    def get(self):
+        # set Content-Encoding manually to avoid automatic gzip encoding
+        self.set_header("Content-Type", "text/plain")
+        self.set_header("Content-Encoding", "gzip")
+        body = "".join("Hello World {}\n".format(i) for i in range(9000))[:149051]
```

Comment:
Add a comment that the length of the input appears to be relevant to triggering the bug, and that this length is the size of the data in the original report (and link to that issue).